R/super.R

```diff
@@ -36,20 +36,80 @@
 super <- function(...) {
   # TODO: (RK) Error on anonymous function calls.
   fn <- as.character(sys.call(-1)[[1]])
-
-  env <- parent.frame(2)
+  if (identical(fn[1], "get")) {
+    if (length(fn) == 1) { # TODO: (RK) Actually parse out super::multi_parent_env
+      stop("super::super does not support ", sQuote("get"), " calls ",
+           "due to internal purposes.")
+    } else {
+      # From the magic get() call below, we extract the function name.
+      fn <- fn[2] 
+      env <- parent.frame()
+      level <- 2 # See comment below.
+    }
+  } else {
+    env   <- parent.frame(2)
+    level <- 7
+  }
 
   first <- TRUE
+  parent_count <- -1
```

Comment:
Starting from the correct parent frame (see `env` above), we must call `parent.env` multiple times until we get to the next available `fn`. **However**, package namespaces [traverse back to the global environment](http://blog.obeautifulcode.com/R/How-R-Searches-And-Finds-Stuff/). If we had a `super` calling chain in packages, we would "drop" back to the first attached package with `fn` each time and go into an infinite loop. Instead, we `hop` (see the `hop` function below) from the package namespace to the package environment, whose parent environment is indeed the next package on the search path.